examples/ai/llamaindex/llamaindex.ipynb

```diff
@@ -0,0 +1,193 @@
+   "cell_type": "markdown",
+   "id": "4f3b89c3",
+   "metadata": {},
+   "source": [
+    "[![Open in Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/supabase/supabase/blob/master/examples/ai/llamaindex/llamaindex.ipynb)"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "id": "2a1f181d-feeb-4b29-aabc-67a75234b92c",
+   "metadata": {},
+   "source": [
+    "# Supabase + LlamaIndex\n",
+    "\n",
+    "In this example we'll use PostgreSQL + pgvectors with [LlamaIndex's Supabase Vector Store](https://gpt-index.readthedocs.io/en/latest/examples/vector_stores/SupabaseVectorIndexDemo.html#setup-openai).\n"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "id": "3cab93f5-10d0-47c5-9f4e-64921461e7e2",
+   "metadata": {},
+   "source": [
+    "## Install Dependencies"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "id": "a41bc3e4-ea52-43aa-9239-a431b49f029e",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!pip install -qU vecs datasets llama_index"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "id": "0026437c",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import logging\n",
+    "import sys\n",
+    "\n",
+    "# Uncomment to see debug logs\n",
+    "# logging.basicConfig(stream=sys.stdout, level=logging.DEBUG)\n",
+    "# logging.getLogger().addHandler(logging.StreamHandler(stream=sys.stdout))\n",
+    "\n",
+    "from llama_index import SimpleDirectoryReader, Document, StorageContext\n",
+    "from llama_index.indices.vector_store import VectorStoreIndex\n",
+    "from llama_index.vector_stores import SupabaseVectorStore\n",
+    "import textwrap"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "id": "7a95a440",
+   "metadata": {},
+   "source": [
+    "# Set up OpenAI\n",
+    "\n",
+    "LlamaIndex uses OpenAI GPT-3 `text-davinci-003` model. Let's store that in an environment variable:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "ca2437d3",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import os\n",
+    "os.environ['OPENAI_API_KEY'] = \"[your_openai_api_key]\""
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "id": "4dbd176f-3b4e-4d41-a72d-1e1affe6ecae",
+   "metadata": {},
+   "source": [
+    "## Load the Dataset\n",
+    "\n",
+    "Let's load a small data set of Paul Graham's essays:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "id": "dc6b0bc2-b95f-4190-bf77-fa2dc57fc247",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Document ID: e2b246de-b306-4615-851c-fe891a862a5f Document Hash: 77ae91ab542f3abb308c4d7c77c9bc4c9ad0ccd63144802b7cbe7e1bb3a4094e\n"
+     ]
+    }
+   ],
+   "source": [
+    "documents = SimpleDirectoryReader('./data').load_data()\n",
```

Comment:
I don't actually know if the `./data` folder will be accessible within Colab?

Reply:
Just tested this here:
https://colab.research.google.com/github/supabase/supabase/blob/ca83ef0/examples/ai/llamaindex/llamaindex.ipynb

And `SimpleDirectoryReader` didn't seem to work. Replaced with `SimpleWebPageReader`:
https://colab.research.google.com/github/supabase/supabase/blob/20ea63f/examples/ai/llamaindex/llamaindex.ipynb

which seems to work (replace `master` with `20ea63f` when calling `load_data()`).

Going to merge to confirm its working on `master`, then we can add a nav link.